### PR TITLE
tweak(scripting/lua): Citizen.Await promise handling

### PR DESCRIPTION
**Pull request by thelindat**
I was having a look at Citizen.Await / promises and noticed some weird hack, despite promises exposing state and value (maybe based on some old code?). I didn't experience any unexpected behaviour with this change to the implementation.

Change to main.js returns the actual error message, rather than an empty object in the Lua runtime.
image
image

### CODE REVIEW

File: data/shared/citizen/scripting/lua/scheduler.lua [hunk 1]
```diff
@@ -849,10 +835,10 @@ local function lazyEventHandler() -- lazy initializer so we don't add an event w
 end
 
 -- Handle an export with multiple return values.
-local function exportProcessResult(resource, k, status, ...)
+local function exportProcessResult(resource, exportName, status, ...)
 	if not status then
 		local result = tostring(select(1, ...))
-		error('An error occurred while calling export ' .. k .. ' of resource ' .. resource .. ' (' .. result .. '), see above for details')
+		error(('An error occured while calling export %s of resource %s\n%s'):format(exportName, resource, result))
```
Comment (blattersturm) [inline data/shared/citizen/scripting/lua/scheduler.lua:841]: How'd you end up introducing a typo?

**Review by blattersturm — CHANGES_REQUESTED** [review]
not gonna merge a typo :D

Odd how the person who PR'd this long ago (#75) did this when apparently the library they imported _did_ make state accessible.

File: data/shared/citizen/scripting/lua/scheduler.lua (hunk 1 above)
Comment (thelindat) [reply data/shared/citizen/scripting/lua/scheduler.lua:841]: 😊 general incompetence

**Comment by blattersturm** [comment]
i don't think our commit categorization likes `refactor` much. we need a `internal` category or so someday but for now `tweak`

**Comment by thelindat** [comment]
> i don't think our commit categorization likes `refactor` much. we need a `internal` category or so someday but for now `tweak`

Amended to tweak. Is there somewhere I can view the acceptable categories?

**Comment by blattersturm** [comment]
> Is there somewhere I can view the acceptable categories?

No, I don't think so. This may be in the changelog code but I'm not sure if this is on the client or server side.

**Review by blattersturm — APPROVED** [review]

**Pull request merged**